api/controllers/groupAdmin.js

```diff
@@ -0,0 +1,314 @@
+                if (err) res.status(500).send(err)
+                res.send({"actions": result.filter(x => x.message.entities != null).length, "members": members, "messages": messages})
+            })
+        });
+        
+    }
+}
+
+exports.getGroupMessages = (mongoCollections) => {
+    return (req, res) => {
+        group_id = parseInt(req.body.group_id)
+        mongoCollections.mongoMessages.aggregate([
+            {
+                $match: {
+                    'message.chat.id': group_id
+                }
+            },
+            {
+                $sort: {
+                    'postedDate': -1
+                }
+            }
+        ], (err, result)=> {
+            if (err) res.status(500).send(err)
+            res.send(result)
+        });
+    }
+}
+
+exports.getGroupMembers = (mongoCollections) => {
+    return (req, res) => {
+        group_id = parseInt(req.body.group_id);
+        mongoCollections.mongoGroupMembers.find({groupId: group_id}).toArray((err, result) => {
+            if (err) res.status(500).send(err)
+            res.send(result)
+        });
+    }
+}
+
+exports.getGroupMessagesYearlyCount = (mongoCollections) => {
+    return (req, res) => {
+        group_id = parseInt(req.body.group_id);
+        mongoCollections.mongoMessages.aggregate([
+            {
+                $match: {
+                    'message.chat.id': group_id
+                }
+            },
+            {
+                $project: {
+                    month: {"$month": '$postedDate'},
+                    messageYear: {"$year": "$postedDate"}
+                }
+            },
+            {
+                $group: {
+                    _id: {month: "$month", year: "$messageYear"},
+                    messages: {$sum: 1}
+                }
+            }
+        ], (err, result) => {
+            if (err) res.status(500).send(err);
+                res.send(result);
+        })
+    }
+}
+
+exports.getGroupMessagesMonthlyCount = (mongoCollections) => {
+    return (req, res) => {
+        group_id = parseInt(req.body.group_id);
+        var month = (new Date()).getMonth() + 1;
+        mongoCollections.mongoMessages.aggregate([
+            {
+                $match: {
+                    'message.chat.id': group_id
+                }
+            },
+            {
+                $project: {
+                    day: {"$dayOfMonth": '$postedDate'},
+                    messageMonth: {"$month": "$postedDate"}
+                }
+            },
+            {
+                $group: {
+                    _id: {day: "$day", month: "$messageMonth"},
+                    messages: {$sum: 1}
+                }
+            },
+            {
+                $match: {
+                    '_id.month': month
+                }
+            }
+        ], (err, result) => {
+            if (err) res.status(500).send(err);
```

Comment:
In most number of callbacks you handle errors in exactly the same way like here. At least you can take it out into single function.